### PR TITLE
Removing `libraft.so` from libcuvs dependencies

### PR DESCRIPTION
**Pull request by cjnolet**
Unfortunately, libraft.so is still a dependency of pylibraft and so it's needed in order to use pylibraft. We'll work on that during the 24.08 release cycle. For now, libraft.so is no longer a required dependency to build `libcuvs` or any of its wrappers.

### CODE REVIEW

File: cpp/include/cuvs/neighbors/common.hpp [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2024, NVIDIA CORPORATION.
+ * Copyright (c) 2022-2023, NVIDIA CORPORATION.
```
Comment (divyegala) [inline cpp/include/cuvs/neighbors/common.hpp:2]: I'm not sure if the copyright check is working correctly in cuvs. Can we file an issue to address it later?

File: cpp/include/cuvs/neighbors/common.hpp [hunk 2]
```diff
@@ -209,7 +280,7 @@ auto make_aligned_dataset(const raft::resources& res, const SrcT& src, uint32_t
 }
 
 /** Parameters for VPQ compression. */
-using vpq_params = cuvs::neighbors::ann::vpq_params;
+using vpq_params = cuvs::neighbors::vpq_params;
```
Comment (divyegala) [inline cpp/include/cuvs/neighbors/common.hpp:283]: Does not look like this line is needed anymore as the class definition is in the same file and same namespace now

**Review by divyegala — COMMENTED** [review]
Minor comments

File: cpp/include/cuvs/neighbors/common.hpp (hunk 2 above)
Comment (cjnolet) [reply cpp/include/cuvs/neighbors/common.hpp:283]: Good catch!

**Review by divyegala — APPROVED** [review]

**Pull request merged**